python/ray/tune/checkpoint_policy.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+
+from ray.tune.result import DONE, TRAINING_ITERATION
+
+logger = logging.getLogger(__name__)
+
+
+class CheckpointPolicy(object):
+    """Interface for Trainable checkpoint policy."""
+
+    def should_checkpoint(self, result):
+        """Determines whether to trigger checkpoint based on the result.
+
+        Args:
+            result (dict): result from last training step.
+
+        Returns:
+            True if should take checkpoint, False otherwise.
+        """
+        raise NotImplementedError
+
+    def checkpoint_score(self, result):
```

Comment:
Should this be more general on this level, i.e. `def garbage_collect` or something like that?

Reply:
yeah i need to refactor this, been busy with other stuff